nav2_costmap_2d/include/nav2_costmap_2d/costmap_2d_ros.hpp

```diff
@@ -88,7 +88,8 @@ class Costmap2DROS : public nav2_util::LifecycleNode
   explicit Costmap2DROS(
     const std::string & name,
     const std::string & parent_namespace,
-    const std::string & local_namespace);
+    const std::string & local_namespace,
+    const bool & use_sim_time = false);
```

Comment:
meaning that I need to change all creations Costmap2DROS to pass an additional use_sim_time parameter?

Reply:
Yes, but to be fair that's only like 3 places and you already did 2 of them - only missing `costmap_2d_node.cpp`. 

Also the `name` constructor on L80 above needs the time too. That's used in more places, but setting a default there would be OK since that's mostly there for generating tests. 

> Wouldn't that be a breaking change for anyone using that class standalone (I don't know how likely that is)?

Yes, but that is what `main` is for, we need to be able to make breaking changes. The fact that we set the `use_sim_time` in launch rather than in the yaml files was already a breaking change enough that this couldn't be backported.

Reply:
Yeah I was talking more about the name-only constructor. Thanks for clarifying that this one can take a default.
Great then, I will try to wrap this up tomorrow

Reply:
> only missing costmap_2d_node.cpp

are you sure? I see the name-constructor being used there